packages/user/Fractals/service/src/tables.rs

```diff
@@ -508,4 +539,115 @@ pub mod tables {
             table.put(&self).expect("failed to save score");
         }
     }
+
+    #[table(name = "BucketTable", index = 4)]
+    #[derive(Default, Fracpack, ToSchema, SimpleObject, Serialize, Deserialize, Debug)]
+    #[graphql(complex)]
+    pub struct Bucket {
+        pub fractal: AccountNumber,
+        pub account: AccountNumber,
+        pub total_deposited: u64,
+        pub total_claimed: u64,
+        pub last_update_principal: u64,
+        pub last_update_timestamp: TimePointSec,
+    }
+
+    impl Bucket {
+        const PPT: u128 = 1_000_000_000_000; // Parts per trillion
+
+        #[primary_key]
+        fn pk(&self) -> (AccountNumber, AccountNumber) {
+            (self.fractal, self.account)
+        }
+
+        pub fn get(fractal: AccountNumber, account: AccountNumber) -> Option<Self> {
+            let table = BucketTable::new();
+            table.get_index_pk().get(&(fractal, account))
+        }
+
+        pub fn get_or_new(fractal: AccountNumber, account: AccountNumber) -> Self {
+            Self::get(fractal, account).unwrap_or(Self {
+                fractal,
+                account,
+                total_claimed: 0,
+                total_deposited: 0,
+                last_update_principal: 0,
+                last_update_timestamp: TimePointSec::from(0),
+            })
+        }
+
+        pub fn get_assert(fractal: AccountNumber, account: AccountNumber) -> Self {
+            check_some(Self::get(fractal, account), "bucket does not exist")
+        }
+
+        fn decay_rate_ppm(&self) -> u128 {
+            Fractal::get_assert(self.fractal).decay_rate_ppm as u128
+        }
+
+        fn balance_still_vesting(&self) -> u64 {
+            if self.last_update_principal == 0 {
+                return 0;
+            }
+            let now = TransactSvc::call().currentBlock().time.seconds();
+            let delta = now
+                .seconds
+                .saturating_sub(self.last_update_timestamp.seconds) as u64;
+            let lambda_t = self.decay_rate_ppm() * (delta as u128);
+            let denominator = Self::PPT.saturating_add(lambda_t);
+            ((self.last_update_principal as u128) * Self::PPT / denominator) as u64
```

Comment:
I think a hyperbola is a poor choice. It has all the negative properties of both linear and exponential curves.